sql/2020/03_Markup/pages_element_count_by_device_and_element_type_present.sql

```diff
@@ -0,0 +1,46 @@
+#standardSQL
+# how many pages contain an element by element and device M241
+# See related: sql/2019/03_Markup/03_02a.sql
+
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+CREATE TEMPORARY FUNCTION get_element_types(element_count_string STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+    if (!element_count_string) return ["ERROR: payload._element_count is missing"]; // 2019 had a few cases
+
+    var element_count = JSON.parse(element_count_string); // should be an object with element type properties with values of how often they are present
+
+    if (Array.isArray(element_count)) return ["ERROR: payload._element_count is an array"];
+    if (typeof element_count != 'object') return ["ERROR: payload._element_count is a " + typeof(element_count)];
+
+    return Object.keys(element_count); 
+} catch (e) {
+    return ["ERROR: "+e.message]; 
+}
+''';
+
+ SELECT
+  _TABLE_SUFFIX AS client,
+  element_type,
```

Comment:
Are all of the element types the same case, or should this be `LOWER(element_type)`?

Reply:
the custom metric returns them all in lowercase.

`let tag = el.tagName.toLowerCase()`